src/components/Map/MapFooter/MapFooter.tsx

```diff
@@ -66,44 +68,77 @@ const Attribution = ({ label, link, title }) => (
   </>
 );
 
+const createAttributionInnerHtml = ({
+  html,
+  text,
+  url,
+}: LayerIndexAttribution) => {
+  if (html) return html;
+
+  if (text && url) return `© <a href="${url}">${text}}</a>`;
+  if (text) return `© ${text}}`;
+
+  return null;
+};
+
 const MapDataLink = () => {
+  const [userLayers] = usePersistedState<Layer[]>('userLayers', []);
   const short = useMediaQuery('(max-width: 500px)');
   const { activeLayers } = useMapStateContext();
   const attributions = uniq(
-    activeLayers.flatMap((layer) =>
-      osmappLayers[layer]
-        ? osmappLayers[layer].attribution
-        : decodeURI(new URL(layer)?.hostname),
-    ),
+    activeLayers.flatMap((layer) => {
+      if (osmappLayers[layer]) return osmappLayers[layer].attribution;
+
+      const uri = decodeURI(new URL(layer)?.hostname);
+
+      try {
+        return userLayers.find(({ url }) => url === layer).attribution || uri;
+      } catch {
+        return uri;
+      }
+    }),
   );
 
-  const nodes = attributions.map((attribution) => {
-    if (attribution === 'maptiler')
-      return (
-        <Attribution
-          key={attribution}
-          label="MapTiler"
-          link="https://www.maptiler.com/"
-          title={<Translation id="map.maptiler_copyright_tooltip" />}
-        />
-      );
-    if (attribution === 'osm')
+  const nodes = attributions.map(
+    (attribution: string | LayerIndexAttribution) => {
+      if (attribution === 'maptiler')
+        return (
+          <Attribution
+            key={attribution}
+            label="MapTiler"
+            link="https://www.maptiler.com/"
+            title={<Translation id="map.maptiler_copyright_tooltip" />}
+          />
+        );
+      if (attribution === 'osm')
+        return (
+          <Attribution
+            key={attribution}
+            label={short ? 'OSM' : 'OpenStreetMap'}
+            link="https://www.openstreetmap.org/"
+            title={<Translation id="map.osm_copyright_tooltip" />}
+          />
+        );
+
+      if (typeof attribution === 'string')
+        return (
+          <span
+            key={attribution}
+            dangerouslySetInnerHTML={{ __html: attribution }} // eslint-disable-line react/no-danger
+          />
+        );
+
+      const { url, text } = attribution;
+      const innerHtml = createAttributionInnerHtml(attribution);
```

Comment:
It seems attribution is ripe for refactoring, this is introducing technical debt.

I suggest always using storing a react node in the attribution, if this is possible. Then here we can construct here a rich object like below, we would make uniq array by keys, and then just output the React nodes. What do you think?
-  `{ key: 'maptiler', node: <MaptilerAttribution/>}`
- `{ key: 'URL/KEY-FROM-LAYER-INDEX', node: <a href="">...</a>}`
- `{ key: 'URL/KEY-FROM-LAYER-INDEX', node: <>...</>}`

We can do it in followup, let me know, if you want to do it yourself or should I. Also if this is a followup it is usually cool to leave a comment saying sth like "TODO this will be refactored in a followup, link-to-issue". So the tech debt doesn't become unmaintainable 🙈

On the other hand if you want to do it here, merge master first, as this file was refactored and moved.